Full deployment

PharkMillups opened the pull request:
First pass at a standalone deployment guide. Still do to:

- Reorg / pull redundant content from other parts of docs 
- Add links to this to top level validator page

We'll also want a full reference page for CLI usage (most of what's on the current "wallet" page) as well as a page that covers all validators rules, primitives, an operations. We'll need separate PRs for this.

Commit:
more additions

dellisny commented:
1) Your miner commands are a bit terse: "miner peer addr". It looks like you are maybe using an alias. 
2) How do you "see" that your Validator is synced and working?
3) How do you check the height of the validator blockchain?

dellisny commented:
Also, it might be good to remind people to open port 2154/tcp on their firewalls.

jthiller commented on docs/mine-hnt/validators/validator-deployment-guide.mdx:
```diff
@@ -0,0 +1,389 @@
+--publish 2154:2154/tcp \
+--name validator \
+--mount type=bind,source=$HOME/validator_data,target=/var/data \
+quay.io/team-helium/validator:miner-val-amd64_val0.1.7
```
```suggestion
quay.io/team-helium/validator:miner-val-amd64_val0.1.8
```
I believe we can reliably point to 0.1.8 now.

kellybyrd replied:
Or maybe just `latest`?

PharkMillups replied:
Switched to `latest-val-amd64`. Thanks.

kellybyrd commented on docs/mine-hnt/validators/validator-deployment-guide.mdx:
````diff
@@ -0,0 +1,389 @@
+The validator directory should be created on a persistent EBS volume (if you're using AWS) or something similar that lives across server lifetimes.
+
+```
+mkdir $HOME/validator_data; source=$HOME/validator_data
````
Should be just `mkdir $HOME/validator_data`

PharkMillups replied:
Indeed. We fixed this yesterday. Thanks.

kellybyrd commented on docs/mine-hnt/validators/validator-deployment-guide.mdx:
```diff
@@ -0,0 +1,389 @@
+After running this, you'll need to input your wallet passphrase to sign the transaction. 
+
+And with that, you're done. Congratulations! You're running a Helium Network Validator. 
+
```
Suggestion:
The first thing someone new is going to ask is: "how do I know it worked?" So, maybe add how long to wait before verifying along with  some commands and links like: 
*  the validators list on the testnet-api 
*  the `in_consensus` cmdline
*  the `info summary` cmdline

PharkMillups replied:
Yup. We should add a section on a few things to test once you've deployed and staked. Care to take a stab at it? I just pushed a new section to this PR where the content can go at the end of the guide.

kellybyrd replied:
Will do this today.

kellybyrd commented:
> Also, it might be good to remind people to open port 2154/tcp on their firewalls.

I think somewhere in this doc (or maybe in the technical requirements?) you should call out ONLY TCP 2154 is needed not any previous ports that people had for hotspots or miners. Reading the discord channel over the last day, several people seem to have assumed 2154 was in addition to 44158.

kellybyrd commented on docs/mine-hnt/validators/validator-deployment-guide.mdx:
```diff
@@ -0,0 +1,389 @@
+
+To successfully deploy a Helium Validator, you'll need to do the following:
+
+* Install the [Helium CLI Wallet](https://github.com/helium/helium-wallet-rs)
```
Based on a question in discord, it made me wonder whether people could assume this means running the wallet on the same machine as the validator. So either here or down in `Create Testnet Wallet` maybe clarify that? 

Or, decide that the target audience understands crypto wallets, dunno.

kellybyrd commented on docs/mine-hnt/validators/validator-deployment-guide.mdx:
```diff
@@ -0,0 +1,389 @@
+To successfully deploy a Helium Validator, you'll need to do the following:
+
+* Install the [Helium CLI Wallet](https://github.com/helium/helium-wallet-rs)
+* Deploy a node on a cloud service like AWS, GCP, Azure, or Digital Ocean. Or deploy a physical server on which to run the validator. [The technical node requirements can be found here](/mine-hnt/validators/requirements/). 
```
"a node" should be something else, I think. The word "node" is often used in Helium discord to talk about things running on the p2p network, so it could be confused with a running hotspot or validator or something else. I think you could say one of:
"an instance" <--- kinda AWS specific, but most folks would get it"
"a VM"
"a VPS" 
"a virtual server" <--- works well with the next sentence which has "a physical server"
"a server"

PharkMillups commented:
> Maybe worthwhile to link to the more complete steps for installing docker on Ubuntu 20.04. Here's a good url: https://docs.docker.com/engine/install/ubuntu/. For most, it would be better to use the latest docker images direct from the docker ppa vs what's in the repo.

Thanks. Link added.

Commit:
a few more additions

Commit:
a few more small tweaks

Commit:
adding link to top level page, moving some words around

Commit:
ordering

allenan commented on docs/mine-hnt/validators/validator-deployment-guide.mdx:
````diff
@@ -0,0 +1,402 @@
+```
+
+```
+sudo apt install g++ -y
````
is there a reason these need to be installed individually on their own line? or could the be added to the mega-apt one-liner above?

PharkMillups replied:
I don't see why we can't add them. I'll update.

allenan commented on docs/mine-hnt/validators/validator-deployment-guide.mdx:
````diff
@@ -0,0 +1,402 @@
+cd miner
+```
+
+And, finally, build the `miner`
````
after this step, and before loading the genesis block, you need to start the miner with `_build/validator/rel/miner/bin/miner start`

PharkMillups replied:
Thanks. Updated.

allenan commented on docs/mine-hnt/validators/validator-deployment-guide.mdx:
````diff
@@ -0,0 +1,402 @@
+
+```
+wget https://snapshots.helium.wtf/genesis.testnet
+miner genesis load </absolute/path/to/genesis/block>
````
we should note that the `miner` binary here is nested within `_build/validator/rel/miner/bin/miner`

PharkMillups replied:
Done. Thanks.

allenan commented on docs/mine-hnt/validators/validator-deployment-guide.mdx:
```diff
@@ -0,0 +1,402 @@
+
+Now that your Validator node is running, the final step in the process is to formally stake TNT to your Validator. As part of the staking process the Validator address needs to both be in the staking transaction and sign the transaction. After a wallet stakes a validator node, the wallet becomes that node’s owner, has control over that validator node, and receives rewards.
+
+First, double check your wallet balance to make sure you have the `10000` TNT required:
```
you'll also need enough TNT to cover the fee, which the extra 5 TNT from the faucet easily covers

PharkMillups replied:
Ack

kellybyrd committed:
Flesh out Verifying Validator Victory (#212)

Added some example commands for validating the a new validator is working and staked.

Commit:
some final tweaks

kent-williams committed:
typo

kent-williams reviewed (approved):
LGTM!
